### ISSUE TITLE
Loaded events trigger on failed HTTP requests

### ISSUE THREAD

**Issue by alex** [servo/servo#8819]
I'm not quite sure which APIs cause this issue, but here is the symptom:
- Browser to https://www.ssllabs.com/ssltest/viewMyClient.html
- See that it claims that SSLv2 and SSLv3 are supported, even though they are not.

This is because some browser API is reporting a success, even though the HTTPS request failed (correctly).

**Comment by jdm** [comment]
I suspect this is caused by the problems that #8678 is intended to fix. That is, presumably the connections are blocked but the resulting error page is treated as a valid network response by the listeners.

**Comment by alex** [comment]
Sounds totally plausible!

**Comment by alex** [comment]
I suspect this is a dupe of #1719, but not positive

**Comment by nox** [comment]
This was fixed, AFAIK.

**Comment by alex** [comment]
@nox I don't think so, https://badssl.com/dashboard/ shows every single result as connected - no error events fired.

**Comment by nox** [comment]
Oh, misread it. Thanks for the quick reaction.
